# 01 — EDA: Audio Features

Análisis exploratorio de las características de audio extraídas del dataset descargado, que fue generado mediante la extracción de datos de la API en 2024.


## 1. Imports y carga de datos 

Se establece la configuración de las librerías que se van a utilizar.

In [1]:
# Librerías de manipulación de datos.
import pandas as pd
import numpy as np


# Librerías de visualización.
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

# Configuración visual
plt.style.use('seaborn-v0_8-whitegrid')
sns.set_palette('husl')
pd.set_option('display.max_columns', None)

A continuación, se carga el dataset con el que se trabajará.

In [2]:
ruta = '../data/raw/dataset.csv'
df_raw = pd.read_csv(ruta)

print(f"Filas: {df_raw.shape[0]:,}")
print(f"Columnas: {df_raw.shape[1]}")
print(f"\nColumnas disponibles:\n{list(df_raw.columns)}")

Filas: 114,000
Columnas: 21

Columnas disponibles:
['Unnamed: 0', 'track_id', 'artists', 'album_name', 'track_name', 'popularity', 'duration_ms', 'explicit', 'danceability', 'energy', 'key', 'loudness', 'mode', 'speechiness', 'acousticness', 'instrumentalness', 'liveness', 'valence', 'tempo', 'time_signature', 'track_genre']


## 2. Estructura del dataset

Buscaremos información sobre la estructura que conforma el conjunto de datos y puede que se realice algún cambio minoritario.

In [3]:
df_raw.head(10)

,Unnamed: 0,track_id,artists,album_name,track_name,popularity,duration_ms,explicit,danceability,energy,key,loudness,mode,speechiness,acousticness,instrumentalness,liveness,valence,tempo,time_signature,track_genre
0,0,5SuOikwiRyPMVoIQDJUgSV,Gen Hoshino,Comedy,Comedy,73,230666,False,0.676,0.4610,1,-6.746,0,0.1430,0.0322,0.000001,0.3580,0.7150,87.917,4,acoustic
1,1,4qPNDBW1i3p13qLCt0Ki3A,Ben Woodward,Ghost (Acoustic),Ghost - Acoustic,55,149610,False,0.420,0.1660,1,-17.235,1,0.0763,0.9240,0.000006,0.1010,0.2670,77.489,4,acoustic
2,2,1iJBSr7s7jYXzM8EGcbK5b,Ingrid Michaelson;ZAYN,To Begin Again,To Begin Again,57,210826,False,0.438,0.3590,0,-9.734,1,0.0557,0.2100,0.000000,0.1170,0.1200,76.332,4,acoustic
3,3,6lfxq3CG4xtTiEg7opyCyx,Kina Grannis,Crazy Rich Asians (Original Motion Picture Sou...,Can't Help Falling In Love,71,201933,False,0.266,0.0596,0,-18.515,1,0.0363,0.9050,0.000071,0.1320,0.1430,181.740,3,acoustic
4,4,5vjLSffimiIP26QG5WcN2K,Chord Overstreet,Hold On,Hold On,82,198853,False,0.618,0.4430,2,-9.681,1,0.0526,0.4690,0.000000,0.0829,0.1670,119.949,4,acoustic
5,5,01MVOl9KtVTNfFiBU9I7dc,Tyrone Wells,Days I Will Remember,Days I Will Remember,58,214240,False,0.688,0.4810,6,-8.807,1,0.1050,0.2890,0.000000,0.1890,0.6660,98.017,4,acoustic
6,6,6Vc5wAMmXdKIAM7WUoEb7N,A Great Big World;Christina Aguilera,Is There Anybody Out There?,Say Something,74,229400,False,0.407,0.1470,2,-8.822,1,0.0355,0.8570,0.000003,0.0913,0.0765,141.284,3,acoustic
7,7,1EzrEOXmMH3G43AXT1y7pA,Jason Mraz,We Sing. We Dance. We Steal Things.,I'm Yours,80,242946,False,0.703,0.4440,11,-9.331,1,0.0417,0.5590,0.000000,0.0973,0.7120,150.960,4,acoustic
8,8,0IktbUcnAGrvD03AWnz3Q8,Jason Mraz;Colbie Caillat,We Sing. We Dance. We Steal Things.,Lucky,74,189613,False,0.625,0.4140,0,-8.700,1,0.0369,0.2940,0.000000,0.1510,0.6690,130.088,4,acoustic
9,9,7k9GuJYLp2AzqokyEdwEw2,Ross Copperman,Hunger,Hunger,56,205594,False,0.442,0.6320,1,-6.770,1,0.0295,0.4260,0.004190,0.0735,0.1960,78.899,4,acoustic


Para comenzar, se eliminará la columna Unamed : 0 y track_id, ya que no aportan información al análisis.

In [4]:
# Trabajamos siempre desde una copia.
df = df_raw.copy()

# Eliminamos la columna 'Unnamed: 0' y 'track_id' que es un índice innecesario.
df.drop(columns=['Unnamed: 0', 'track_id'], inplace=True)
print(f'Columnas eliminadas correctamente!')

Columnas eliminadas correctamente!


Mostramos los tipos de datos que conforman el dataset, los valores nulos y los duplicados.

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 114000 entries, 0 to 113999
Data columns (total 19 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   artists           113999 non-null  object 
 1   album_name        113999 non-null  object 
 2   track_name        113999 non-null  object 
 3   popularity        114000 non-null  int64  
 4   duration_ms       114000 non-null  int64  
 5   explicit          114000 non-null  bool   
 6   danceability      114000 non-null  float64
 7   energy            114000 non-null  float64
 8   key               114000 non-null  int64  
 9   loudness          114000 non-null  float64
 10  mode              114000 non-null  int64  
 11  speechiness       114000 non-null  float64
 12  acousticness      114000 non-null  float64
 13  instrumentalness  114000 non-null  float64
 14  liveness          114000 non-null  float64
 15  valence           114000 non-null  float64
 16  tempo             11

In [6]:
print(f'\nValores nulos por columna:')
nulos = df.isnull().sum()
print(nulos[nulos > 0] if nulos.sum() > 0 else 'No hay valores nulos en el dataset.')

print(f'\nDuplicados:')
dups = df.duplicated().sum()
print(f'Número de canciones duplicadas: {dups}')



Valores nulos por columna:
artists       1
album_name    1
track_name    1
dtype: int64

Duplicados:
Número de canciones duplicadas: 577


Estadísticas descriptivas de las variables numéricas

In [7]:
# Escogemos valores numéricos
audio_feat = ['danceability', 'energy', 'loudness', 'speechiness','acousticness', 'instrumentalness', 'liveness','valence', 'tempo', 'popularity']

df[audio_feat].describe().round(3)

,danceability,energy,loudness,speechiness,acousticness,instrumentalness,liveness,valence,tempo,popularity
count,114000.000,114000.000,114000.000,114000.000,114000.000,114000.000,114000.000,114000.000,114000.000,114000.000
mean,0.567,0.641,-8.259,0.085,0.315,0.156,0.214,0.474,122.148,33.239
std,0.174,0.252,5.029,0.106,0.333,0.310,0.190,0.259,29.978,22.305
min,0.000,0.000,-49.531,0.000,0.000,0.000,0.000,0.000,0.000,0.000
25%,0.456,0.472,-10.013,0.036,0.017,0.000,0.098,0.260,99.219,17.000
50%,0.580,0.685,-7.004,0.049,0.169,0.000,0.132,0.464,122.017,35.000
75%,0.695,0.854,-5.003,0.084,0.598,0.049,0.273,0.683,140.071,50.000
max,0.985,1.000,4.532,0.965,0.996,1.000,1.000,0.995,243.372,100.000


Tras comprobar que hay 577 valores duplicados y 1 valor nulo, procedemos a eliminarlos.

In [8]:
# Eliminar filas con valores nulos.
df = df.dropna(subset=['artists', 'album_name', 'track_name'])

# Eliminar duplicados.
df = df.drop_duplicates()

# Reset del índice.
df = df.reset_index(drop=True)

print(f"Dataset limpio: {df.shape[0]:,} filas, {df.shape[1]} columnas")

Dataset limpio: 113,422 filas, 19 columnas
